### PR TITLE
chore(config): make initial_permits and batched_permits configurable

### PR DESCRIPTION
**Pull request by lmatz**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?
As
> Streaming exchange buffer size aka. permits has significant impact on performance, which is much beyond our expectations.

make these two variable easier to tune.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### CODE REVIEW

File: src/common/src/config.rs [hunk 1]
```diff
@@ -360,6 +360,16 @@ pub struct DeveloperConfig {
     /// The maximum size of the chunk produced by executor at a time.
     #[serde(default = "default::developer::stream_chunk_size")]
     pub stream_chunk_size: usize,
+
+    /// The initial permits that a channel holds, i.e., the maximum row count can be buffered in
+    /// the channel.
+    #[serde(default = "default::developer::stream_initial_permits")]
+    pub stream_initial_permits: usize,
```
Comment (BugenZhao) [inline src/common/src/config.rs:367]: How about naming it `stream_exchange_initial_permits`?

File: src/stream/src/executor/integration_tests.rs [hunk 2]
```diff
@@ -74,7 +77,7 @@ async fn test_merger_sum_aggr() {
             1,
         )
         .unwrap();
-        let (tx, rx) = channel();
+        let (tx, rx) = channel(INITIAL_PERMITS, BATCHED_PERMITS);
```
Comment (BugenZhao) [inline src/stream/src/executor/integration_tests.rs:80]: How about adding a `channel_for_test` function for this?

**Review by BugenZhao — COMMENTED** [review]
Rest LGTM!

File: src/common/src/config.rs (hunk 1 above)
Comment (lmatz) [reply src/common/src/config.rs:367]: cool!

File: src/stream/src/executor/integration_tests.rs (hunk 2 above)
Comment (lmatz) [reply src/stream/src/executor/integration_tests.rs:80]: Sure👌

**Review by BugenZhao — APPROVED** [review]

**Pull request merged**